mojo/core/node_controller.cc

```diff
@@ -655,8 +656,20 @@ void NodeController::DropPeer(const ports::NodeName& node_name,
     auto it = peers_.find(name);
 
     if (it != peers_.end()) {
+      auto channel_has_one_ref = it->second.get()->HasOneRef();
+
+      // Since some amount of work happens after this point, there's no
+      // guarantee that this assert will fire even if there is a thread
+      // interleaving issue present.
+      //
+      // Based on the comment above, it seems likely that we might be able to
```

Comment:
Nitpick: Are you referring to the comment you mention in the Linear comment [here](https://linear.app/replay/issue/RUN-1050#comment-e25ee2d1)? Do you want to reference it directly? (Else, this is hard to understand)

Reply:
There's a note above (in the original Chromium source) that justifies the copy of the `NodeName` that also seems to suggest that we might be able to make the behavior of the function more consistent by extending the lifetime of the channel.

I also added some context to the linear issue so I'll remove this from the code comment.